src/syscheckd/db/include/fimDBHelper.hpp

```diff
@@ -161,16 +162,18 @@ namespace FIMDBHelper
         update["table"] = tableName;
         update["data"] = {item};
         bool error = false;
-        auto callback {
-            [&error](ReturnTypeCallback type, const nlohmann::json &)
+        auto callback
+        {
+            [&error](ReturnTypeCallback type, const nlohmann::json&)
             {
                 if (type == ReturnTypeCallback::DB_ERROR)
                 {
                     error = true;
                 }
             }
         };
-        if(error)
+
+        if (error)
```

Comment:
this does not make sense, the callback will be executed after this line of code, always "error" will be false.

We could throw an exception on line 171 and catch it in the caller.